montrek-software/montrek
# 375 complete generation for simple table

# Changes
- add templates and management commands for generating a new table with CRUD functionality and tests

# Test 
- create a new app: `python manage.py startapp newapp`
- create table: `python manage.py generate_table newapp foo`
- add `newapp` to `INSTALLED_APPS` in `.env`
- add `newapp.foo` to `NAVBAR_APPS` in `.env`
- make migrations: `python manage.py makemigrations`
- migrate: `python manage.py migrate`
- run tests: `python manage.py test newapp.tests`
- (re)start server: `python manage.py runserver`
- check new table:
<img width="1288" alt="image" src="https://github.com/montrek-software/montrek/assets/30405217/5a88a39e-17ae-4d51-b5b8-3050d71f985f">

- add another table: `python manage.py generate_table newapp bar`
- make migrations: `python manage.py makemigrations`
- migrate: `python manage.py migrate`
- adjust `newapp/urls/__init__.py` (maybe this can also be handled automatically in the future):

```python
from .bar_urls import urlpatterns as bar_urlpatterns
from .foo_urls import urlpatterns as foo_urlpatterns

urlpatterns = foo_urlpatterns + bar_urlpatterns
```

- run tests: `python manage.py test newapp.tests`
- add `newapp.bar` to `NAVBAR_APPS` in `.env`
- restart server
- check new table:
<img width="1249" alt="image" src="https://github.com/montrek-software/montrek/assets/30405217/2a93204c-2f40-4b83-9641-d381facb9cb8">

# Cleanup
- Remove satellite tables from db:
<img width="561" alt="image" src="https://github.com/montrek-software/montrek/assets/30405217/a800ad61-dbb7-41c7-9b89-f07a3c74adf6">

- Remove hub tables from db:
<img width="478" alt="image" src="https://github.com/montrek-software/montrek/assets/30405217/b4bf2cd8-2a37-4621-8d65-2c651add8d85">

- Remove `newapp`, `newapp.foo`, `newapp.bar` from `.env`
- Remove app directory: `rm -r newapp`
